File: day07/day07-input-output-01-python/example1.py
import pulumi_aws as aws

# 建立 Zone
zone = aws.route53.Zone("example", name="example.com.")

www = aws.route53.Record("example",
                         # 將 zone 的 output 屬性當作 record 的 input 屬性
                         zone_id=zone.zone_id,
                         name="www.example.com.",
                         type="TXT",
                         records=["Hello, World"])

File: day07/day07-input-output-01-golang/example1.go
package example1

import (
	"github.com/pulumi/pulumi-aws/sdk/v6/go/aws/route53"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func example1() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		// 建立 Zone
		zone, err := route53.NewZone(ctx, "example", &route53.ZoneArgs{
			Name: pulumi.String("example.com."),
		})
		if err != nil {
			return err
		}

		www, err := route53.NewRecord(ctx, "example", &route53.RecordArgs{
			// 將 zone 的 output 屬性當作 record 的 input 屬性
			ZoneId: zone.ZoneId,
			Name:   pulumi.String("www.example.com."),
			Type:   pulumi.String("TXT"),
			Records: pulumi.StringArray{
				pulumi.String("Hello, World"),
			},
		})
		return nil
	})
}
